### ISSUE TITLE
Why does System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription report .NET Core 4.0.0.0?

### ISSUE THREAD

**Issue by shanselman** [dotnet/corefx#9725]
Why does System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription report .NET Core 4.0.0.0?

/cc @DamianEdwards

**Comment by ericstj** [comment]
/cc @weshaggard @eerhardt 
Original discussion here: https://github.com/dotnet/corefx/pull/5210

**Comment by Priya91** [comment]
The version returned is ```"{FrameworkName} {typeof(object).GetTypeInfo().Assembly.GetName().Version}"```, it's the build version of mscorlib.dll.. 

Initially we had the frameworkdescription just return the stack used, and then ASP.NET had a request to return some version info. There wasn't clear guidelines on what the version should be, runtime or framework or the version in the brand name. And we decided to go with this.

**Comment by DamianEdwards** [comment]
That looks much nicer. Couldn't we make this API implement differently in UWP vs. dotnet.exe host, whether by runtime late-binding or #ifdef, etc.

**Comment by ericstj** [comment]
> because UWP doesn't support the FileVersionInfo stuff.

I added support for that.

**Comment by borgdylan** [comment]
Small Note: The .NET Core version of S.R.IS.RI gets injected into the compilation even if targeting Mono/.NET due to the netstandard1.1 compatibility. This does not only report ".NETCore,Version=4.0.0.0"  as described above but it also produces incorrect output when targeting Mono, since the System.Native.so library has to be provided separately.

**Comment by ericstj** [comment]
@borgdylan that issue is discussed here: https://github.com/dotnet/corefx/issues/9012.

**Comment by borgdylan** [comment]
Please, could this API still be able to be included when using full framework and mono. Mono is currently tracking net461 i.e netstandard1.4.

**Comment by evil-shrike** [comment]
Why the issue is closed? For .NET Core 2.1 `RuntimeInformation.FrameworkDescription` still reports incorrect version: ".NET Core: 4.6.26515.07"

```
>dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300\

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba
```

**Comment by danmosemsft** [comment]
Reopening for discussion if needed.

**Comment by SimonCropp** [comment]
note that the above workaround of 

```
 var runtimeinfoVersion = typeof(RuntimeInformation).GetTypeInfo().Assembly.GetCustomAttributes().OfType<AssemblyInformationalVersionAttribute>().Single().InformationalVersion;
```

which used to return 
```
1.0.24212.01. Commit Hash: 9688ddbb62c04189cac4c4a06e31e93377dccd41
```

now on 2.1 returns 
```
4.6.26011.01 built by: dlab-DDVSOWINAGE026. 
```

so no longer useful

**Comment by jkoritzinsky** [comment]
Fixed by #35516
